projects/ngx-charts-on-fhir/src/lib/fhir-data/fhir-data.service.ts

```diff
@@ -94,6 +99,85 @@ export class FhirDataService {
       return teardownLogic;
     });
   }
+
+  addPatientData(resource: fhirclient.FHIR.Resource) {
+    return this.client?.create(resource);
+  }
+
+  createResourceData(reportBPValue: BloodPressure): fhirclient.FHIR.Resource {
+    return {
+      "resourceType": "Observation",
+      "status": "final",
+      "category": [
+        {
+          "coding": [
+            {
+              "system": "http://terminology.hl7.org/CodeSystem/observation-category",
+              "code": "vital-signs",
+              "display": "vital-signs"
+            }
+          ]
+        }
+      ],
+      "code": {
+        "coding": [
+          {
+            "system": "http://loinc.org",
+            "code": "85354-9",
+            "display": "Blood Pressure"
+          }
+        ],
+        "text": "Blood Pressure"
+      },
+      "subject": {
+        "reference": `Patient/${this.client?.patient.id}`
+      },
+      "encounter": {
+        "reference": `Encounter/${this.client?.encounter.id}`
+      },
```

Comment:
This has the same encounter issue that you fixed in your other PR. Why is this PR duplicating that code? There will be merge conflicts.